RELEASE_WORKFLOW.md

```diff
@@ -0,0 +1,36 @@
+# Release Workflow
+
+This repository uses a system similar to git-flow and has 4 types of branches.
+
+- `main`
+- `release-v{X.X}` (Contains all changes for a given release)
+- `develop` (Contains reviewed/approved changes, may contain changes that will not go into the same release)
+- Work branches dedicated to a specification addition/change.
+
+## Add/Change a specification
+
+- Create a new branch starting from an up-to-date develop.
+
+## Release Process
+
+- Create a `release-v{X.X}` branch starting from an up-to-date `main`.
+
+### Insert a change to a release
+
+- Merge squash the pull-request modifying/adding a specification in `develop`, if it has not been done earlier.
+- Move to the `release-v{X.X}` branch and cherry pick the previous commit created during the merge squash operation.
+
+### Ship a release
+
+- Merge squash the `release-v{X.X}` branch pull-request into `main`.
+- Pull `main` into `develop`.
+
+### Rebase current work branches on develop
+
+- Move to the given `branch_name`
+- `git pull origin branch_name`
+- `git rebase origin/develop`
```

Comment:
Yes, this is how I'm doing the rebase onto develop for branches that need to be up-to-date on develop. (Every working branch in fact)

Reply:
Oh `branch_names` are other `release-vX.X` branches?

For me, either you just write "Rebase other branches starting from `develop`" or you put detailed commands.
For me
- the `git push --force origin branch` is missing
- the "fix the potential git conflict" is missing
- to be consistent, write `git checkout branch_name` instead of "Move to ..."